app/models/ems_refresh.rb

```diff
@@ -128,6 +134,31 @@ def self.queue_merge(targets, ems)
     end
   end
 
+  def self.queue_merge_sync(targets, queue_options)
+    task = MiqTask.create(
+      :name    => 'Refresh',
+      :userid  => 'system',
+      :state   => MiqTask::STATE_QUEUED,
+      :status  => MiqTask::STATUS_OK,
+      :message => "Queued the action: ['Refresh'] being run for user: ['system']"
+    )
+
+    # Set the callback for this task to set the status based on the results of the actions
+    cb = {:class_name => task.class.name, :instance_id => task.id, :method_name => :queue_callback, :args => ['Finished']}
+
+    MiqQueue.put_or_update(queue_options) do |msg, item|
+      targets = msg.nil? ? targets : (msg.args[0] | targets)
+      item.merge(
+        :args         => [targets],
+        :msg_timeout  => queue_timeout,
+        :task_id      => nil,
+        :miq_callback => cb
```

Comment:
I don't think we can use callbacks based on my comment here: https://github.com/ManageIQ/manageiq/pull/10176#issuecomment-237906310

In short, if multiple queue_merge_syncs are called, we need multiple callbacks, but you can only have 1 callback on the queue record, but we do a put_or_update.